### PR TITLE
chore: initialize extension boilerplate

### PR DESCRIPTION
**Pull request by kevineger**
ext_boiler

Generated largely from `yo code` as per https://code.visualstudio.com/api/get-started/your-first-extension. Modifications include:

- Minimal `.gitignore`.
- Launch config for debugging the integration test.
- Stripped boilerplate comments.
- Added `test:xvfb` script and modified `test` to better support running tests in headless environments. Needed when working on this codebase using the remote toolkit in VS Code (e.g. SSH).
- Placeholder _Hello World!_ command.

Modifications to the defaults to follow (e.g. TS config, ES lint). For now, simply includes a _Hello World_ command to test the extension.

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -0,0 +1,10 @@
+{
+	"files.exclude": {
+		"out": false
+	},
+	"search.exclude": {
+		"out": true
+	},
+	// Turn off tsc task auto detection since we have the necessary tasks as npm scripts.
```
Comment (katlyn-edwards) [inline .vscode/settings.json:8]: (IDK how to make "resolved" comments - no action required)
I bothered Kevin about this over chat, I guess comments are allowed if it's JSONC - https://changelog.com/news/jsonc-is-a-superset-of-json-which-supports-comments-6LwR 

(TIL!)

File: src/extension.ts [hunk 2]
```diff
@@ -0,0 +1,15 @@
+import * as vscode from 'vscode';
+
+// Called when the extension is activated.
+export function activate(context: vscode.ExtensionContext) {
+	console.log('Colab is active!');
```
Comment (katlyn-edwards) [inline src/extension.ts:5]: not intended to start a fight:
I was curious why this felt wildly indented, and I think it's using a "Tab" character, while some of the other files use spaces. (Maybe the Tab character is really wide in GitHub?)

No action needed, just vibes

File: src/test/runTest.ts [hunk 3]
```diff
@@ -0,0 +1,20 @@
+import * as path from 'path';
+
+import { runTests } from '@vscode/test-electron';
+
+async function main() {
+    try {
```
Comment (katlyn-edwards) [inline src/test/runTest.ts:6]: Just to note: this one is using spaces? I don't care what we pick, but it'd be nice to be consistent.

File: tsconfig.json [hunk 4]
```diff
@@ -0,0 +1,13 @@
+{
+	"compilerOptions": {
```
Comment (katlyn-edwards) [inline tsconfig.json:2]: re: tabs/spaces this inconsistency is also present across the JSON files -- some are big tabs and some appear to be spaces. Again, no prefs, or if there's a reason to deviate, that's also cool, more of a curiosity.

File: src/extension.ts (hunk 2 above)
Comment (kevineger) [reply src/extension.ts:5]: All vibes welcome 🤙.

A couple things:

1. In GitHub, you can [manage your tab size rendering preference](https://docs.github.com/en/account-and-profile/setting-up-and-managing-your-personal-account-on-github/managing-user-account-settings/managing-your-tab-size-rendering-preference).
2. In a subsequent PR, I'll enforce tabs vs. spaces. Haven't looked in detail, but I believe this is best setup with ESLint, optionally with Prettier. Commits can be guarded to enforce formatting and linting with a GitHub action that runs before merging `main`.

File: src/test/runTest.ts (hunk 3 above)
Comment (kevineger) [reply src/test/runTest.ts:6]: See [other comment](https://github.com/googlecolab/colab-vscode/pull/1#discussion_r1861105031). Enforcement to follow.

File: tsconfig.json (hunk 4 above)
Comment (kevineger) [reply tsconfig.json:2]: See [other comment](https://github.com/googlecolab/colab-vscode/pull/1#discussion_r1861105031). Enforcement to follow.

**Comment by kevineger** [comment]
> Wow it's been a _while_ since I've seen Yeoman 0_o.
> 
> You say _"Generated largely from `yo code`"_. If easy, do you mind summarizing the changes you made from what Yeoman generated? I'm less interested in Yeoman's auto-generated code and more interested in the stuff we manually added/changed. Regardless, looks like a bunch of boring standard boilerplate stuff, easy to iterate on in the future, approved.

Sure thing, updated [main PR comment](https://github.com/googlecolab/colab-vscode/pull/1#issue-2696274538).

**Pull request merged**